### PR TITLE
Implement Stripe Authorization API

### LINKED ISSUE #4062: Implement StripeAuthorization API with permissions
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-orga-server

### PR DESCRIPTION
**Pull request by poush**
Fixes #4062 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Added Stripe-Authorization API

- `/strip-authorization` [GET, POST]
- `/events/identifier/stripe` [ GET ]

### CODE REVIEW

File: app/api/__init__.py [hunk 1]
```diff
@@ -359,3 +361,12 @@
 api.route(CustomFormDetail, 'custom_form_detail', '/custom-forms/<int:id>')
 api.route(CustomFormRelationshipRequired, 'custom_form_event',
           '/custom-forms/<int:id>/relationships/event')
+
+# Stripe Authorization API
+api.route(StripeAuthorizationListPost, 'stripe_authorization_list_post', '/stripe-authorization')
+api.route(StripeAuthorizationList, 'stripe_authorization_list', '/events/<int:event_id>/stripe',
```
Comment (niranjan94) [inline app/api/__init__.py]: There will only be one per event.

File: app/api/__init__.py [hunk 2]
```diff
@@ -359,3 +361,12 @@
...
+          '/events/<event_identifier>/stripe')
+api.route(StripeAuthorizationDetail, 'stripe_authorization_detail',  '/stripe-authorization/<int:id>',
```
Comment (niranjan94) [inline app/api/__init__.py]: And call it 'stripe-authorization' here too

File: app/api/__init__.py (hunk 1 above)
Comment (niranjan94) [inline app/api/__init__.py]: And we'll need `POST /v1/stripe-authorization` with a compulsory event relationship.

File: app/api/__init__.py (hunk 2 above)
Comment (poush) [inline app/api/__init__.py]: You mean to there will be no any `<int:id>`

Comment (niranjan94) [inline app/api/__init__.py]: I meant to comment on the line above.

File: app/api/__init__.py (hunk 1 above)
Comment (poush) [inline app/api/__init__.py]: Done

File: app/api/__init__.py (hunk 2 above)
Comment (poush) [inline app/api/__init__.py]: Done :)

Comment (niranjan94) [inline app/api/__init__.py]: @poush this isn't done yet ?

Comment (poush) [inline app/api/__init__.py]: @niranjan94 as you told, I changed it to "/events/<int:event_id>/stripe-authorization"

File: app/api/__init__.py [hunk 3]
```diff
@@ -359,3 +361,12 @@
 api.route(CustomFormDetail, 'custom_form_detail', '/custom-forms/<int:id>')
 api.route(CustomFormRelationshipRequired, 'custom_form_event',
           '/custom-forms/<int:id>/relationships/event')
+
+# Stripe Authorization API
+api.route(StripeAuthorizationListPost, 'stripe_authorization_list_post', '/stripe-authorization')
+api.route(StripeAuthorizationList, 'stripe_authorization_list', '/events/<int:event_id>/stripe-authorization',
+          '/events/<event_identifier>/stripe-authorization')
+api.route(StripeAuthorizationDetail, 'stripe_authorization_detail',  '/stripe-authorization/<int:id>',
+          '/events/<int:event_id>/stripe', '/events/<event_identifier>/stripe')
```
Comment (niranjan94) [inline app/api/__init__.py]: @poush but it still says `stripe` here ...

Comment (poush) [inline app/api/__init__.py]: Sorry for taking long 😅 got some college work to complete.

**Comment by niranjan94** [comment]
A secondary review would be nice @fossasia/open-event-server-devs

**Pull request merged**